drivers/painter/generic/qp_rgb565_surface.c

```diff
@@ -0,0 +1,279 @@
+    rgb565_surface_painter_device_t *surface = (rgb565_surface_painter_device_t *)driver;
+    stream_pixdata(surface, (const uint16_t *)pixel_data, native_pixel_count);
+    return true;
+}
+
+// Pixel colour conversion
+static bool qp_rgb565_surface_palette_convert_rgb565_swapped(painter_device_t device, int16_t palette_size, qp_pixel_t *palette) {
+    for (int16_t i = 0; i < palette_size; ++i) {
+        RGB      rgb      = hsv_to_rgb_nocie((HSV){palette[i].hsv888.h, palette[i].hsv888.s, palette[i].hsv888.v});
+        uint16_t rgb565   = (((uint16_t)rgb.r) >> 3) << 11 | (((uint16_t)rgb.g) >> 2) << 5 | (((uint16_t)rgb.b) >> 3);
+        palette[i].rgb565 = BYTE_SWAP(rgb565);
+    }
+    return true;
+}
+
+// Append pixels to the target location, keyed by the pixel index
+static bool qp_rgb565_surface_append_pixels_rgb565(painter_device_t device, uint8_t *target_buffer, qp_pixel_t *palette, uint32_t pixel_offset, uint32_t pixel_count, uint8_t *palette_indices) {
+    uint16_t *buf = (uint16_t *)target_buffer;
+    for (uint32_t i = 0; i < pixel_count; ++i) {
+        buf[pixel_offset + i] = palette[palette_indices[i]].rgb565;
+    }
+    return true;
+}
+
+const struct painter_driver_vtable_t rgb565_surface_driver_vtable = {
+    .init            = qp_rgb565_surface_init,
+    .power           = qp_rgb565_surface_power,
+    .clear           = qp_rgb565_surface_clear,
+    .flush           = qp_rgb565_surface_flush,
+    .pixdata         = qp_rgb565_surface_pixdata,
+    .viewport        = qp_rgb565_surface_viewport,
+    .palette_convert = qp_rgb565_surface_palette_convert_rgb565_swapped,
+    .append_pixels   = qp_rgb565_surface_append_pixels_rgb565,
+};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Comms vtable
+
+static bool qp_rgb565_surface_comms_init(painter_device_t device) {
+    // No-op.
+    return true;
+}
+static bool qp_rgb565_surface_comms_start(painter_device_t device) {
+    // No-op.
+    return true;
+}
+static void qp_rgb565_surface_comms_stop(painter_device_t device) {
+    // No-op.
+}
+uint32_t qp_rgb565_surface_comms_send(painter_device_t device, const void *data, uint32_t byte_count) {
+    // No-op.
+    return byte_count;
+}
+
+struct painter_comms_vtable_t rgb565_surface_driver_comms_vtable = {
+    // These are all effective no-op's because they're not actually needed.
+    .comms_init  = qp_rgb565_surface_comms_init,
+    .comms_start = qp_rgb565_surface_comms_start,
+    .comms_stop  = qp_rgb565_surface_comms_stop,
+    .comms_send  = qp_rgb565_surface_comms_send};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Factory function for creating a handle to an rgb565 surface
+
+painter_device_t qp_make_rgb565_surface(uint16_t panel_width, uint16_t panel_height, void *buffer) {
```

Comment:
```suggestion
painter_device_t qp_rgb565_make_surface(uint16_t panel_width, uint16_t panel_height, void *buffer) {
```

Switching this around would be more consistent with the rest and how you make displays.